docs/writing_rules/examples.md

```diff
@@ -0,0 +1,232 @@
+patterns:
+- pattern-not: subprocess.call("...")
+- pattern: subprocess.call(...)
+```
+
+```python
+import subprocess
+import sys
+
+subprocess.call("echo 'hello'") # Doesn't match here anymore!
+subprocess.call("grep -R {} .".format(sys.argv[1])) # Matches here
+subprocess.call("grep -R {} .".format(sys.argv[1]), shell=True) # Matches here
+subprocess.call("grep -R {} .".format(sys.argv[1]), shell=True, cwd="/home/user") # Matches here
+subprocess.run("grep -R {} .".format(sys.argv[1]), shell=True) # Doesn't match here
+```
+
+https://semgrep.live/v8X8
+
+**Look explicitly for dangerous keyword arguments.** You may want to match only when [certain keyword arguments are present](https://docs.python.org/3/library/subprocess.html#security-considerations). We can write keyword arguments just like in Python into our pattern. Combined with the ellipsis operator, this pattern will match if `shell=True` appears at the end of the sequence of arguments.
+
+```yaml
+patterns:
+- pattern-not: subprocess.call("...")
+- pattern: subprocess.call(..., shell=True)
+```
+
+```python
+import subprocess
+import sys
+
+subprocess.call("echo 'hello'") # Doesn't match
+subprocess.call("grep -R {} .".format(sys.argv[1])) # Doesn't match here anymore!
+subprocess.call("grep -R {} .".format(sys.argv[1]), shell=True) # Matches here
+subprocess.call("grep -R {} .".format(sys.argv[1]), shell=True, cwd="/home/user") # Oops! We don't match here anymore either!
+subprocess.run("grep -R {} .".format(sys.argv[1]), shell=True) # Doesn't match here
+```
+
+https://semgrep.live/d8J6
+
+Semgrep will match `(..., shell=True)` only when `shell=True` is the last argument. To fix this, we can use the ellipsis operator on both sides of `shell=True`.
+
+```yaml
+patterns:
+- pattern-not: subprocess.call("...")
+- pattern: subprocess.call(..., shell=True, ...)
+```
+
+```python
+import subprocess
+import sys
+
+subprocess.call("echo 'hello'") # Doesn't match
+subprocess.call("grep -R {} .".format(sys.argv[1])) # Doesn't match
+subprocess.call("grep -R {} .".format(sys.argv[1]), shell=True) # Matches here
+subprocess.call("grep -R {} .".format(sys.argv[1]), shell=True, cwd="/home/user") # Matches here too!
+subprocess.run("grep -R {} .".format(sys.argv[1]), shell=True) # Doesn't match here
+```
+
+https://semgrep.live/ZqKW
+
+**Bonus: Match any `subprocess` function with `shell=True`.** As you probably noticed, `subprocess.run` is subject to the same issue as `subprocess.call`. `subprocess.run` [was made available in Python 3.5](https://docs.python.org/3/library/subprocess.html#older-high-level-api). We can match both `subprocess.call` and `subprocess.run` by using **metavariables**. Metavariables let you match any code expression. To use metavariables in Semgrep, use the dollar sign as a prefix and all capital letters. In this example, we will use `subprocess.$FUNC`. The name can be anything -- it's just a like a variable in a normal language and will "hold" the expression it matches.
```

Comment:
Consider splitting this into 2 paragraphs to make it more readable - add newline before "Metavariables let you match any code expression."

Consider adding at the end of second paragraph: 

> To learn more about metavariables, see the [primary docs](../pattern-features.md#metavariables).